### PR TITLE
test: add a unit test for `gitlab/__main__.py`

### PR DESCRIPTION
**Pull request by JohnVillalovos**

### CODE REVIEW

**Review by nejch — COMMENTED** [review]
Hmm, I agree with what was said in other places that not sure it makes sense to test this at unit test level ([1](https://stackoverflow.com/questions/5850268/how-to-test-or-mock-if-name-main-contents), [2](https://stackoverflow.com/questions/49412706/unit-testing-main-py), [3](https://stackoverflow.com/questions/51736864/how-to-test-command-line-applications-in-python/51736889#51736889)), and we already have a test for this entrypoint (https://github.com/python-gitlab/python-gitlab/blob/main/tests/functional/cli/test_cli.py#L21-L23).

But if we do, would [runpy](https://docs.python.org/3/library/runpy.html) be enough for this?

File: tests/unit/test__main__.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    # Make the `name` be `__main__` so the if condition will be met
+    spec = importlib.util.spec_from_file_location(
+        name="__main__", location=main_py_path
+    )
+    module = importlib.util.module_from_spec(spec=spec)
+    sys.modules["gitlab.__main__"] = module
+    with pytest.raises(SystemExit) as exc:
+        spec.loader.exec_module(module)
+    caught_exception = exc.value
+    assert caught_exception.code == 2
```
[the remark this answers is not in the document]
Comment (JohnVillalovos) [reply tests/unit/test__main__.py:25]: The reason for the variable is to help with readability. Most people are not likely to know that `value` is the caught exception when reading the code. By creating the variable it helps the reader to understand that it is the exception.

**Comment by JohnVillalovos** [comment]
> Hmm, I agree with what was said in other places that not sure it makes sense to test this at unit test level ([1](https://stackoverflow.com/questions/5850268/how-to-test-or-mock-if-name-main-contents), [2](https://stackoverflow.com/questions/49412706/unit-testing-main-py), [3](https://stackoverflow.com/questions/51736864/how-to-test-command-line-applications-in-python/51736889#51736889)), and we already have a test for this entrypoint (https://github.com/python-gitlab/python-gitlab/blob/main/tests/functional/cli/test_cli.py#L21-L23).
> 
> But if we do, would [runpy](https://docs.python.org/3/library/runpy.html) be enough for this?

`runpy` looks interesting! Let me read more about it tomorrow. Thanks.

File: tests/unit/test__main__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (nejch) [reply tests/unit/test__main__.py:25]: Ok, although IMO we could also use a clearer context manager like `as excinfo:` to signify that and then use `excinfo.value.code` as this is standard exception info same as returned by [`sys.exc_info()`](https://docs.python.org/3/library/sys.html#sys.exc_info). 

E.g. https://github.com/PyCQA/pylint/blob/9b4d46615e22313e446e2ba15e98c0f1d89ccf13/tests/config/test_argparse_config.py#L28-L47 (I see pylint prefers your class-based tests :grin: )

**Comment by nejch** [comment]
Do you think we still need this @JohnVillalovos? We have 100% coverage on `gitlab/__main__.py` now:
https://app.codecov.io/gh/python-gitlab/python-gitlab/blob/main/gitlab/__main__.py

(see also SO discussion linked above)

**Comment by JohnVillalovos** [comment]
> Do you think we still need this @JohnVillalovos? We have 100% coverage on `gitlab/__main__.py` now: https://app.codecov.io/gh/python-gitlab/python-gitlab/blob/main/gitlab/__main__.py
> 
> (see also SO discussion linked above)

Probably fine to not have it. I just thought it was kind of cool 🤓

**Pull request closed without merging**